### DISCUSSION TITLE
instersection nested shell

### DISCUSSION THREAD

**Discussion by ManTouCheer** [shapely/shapely#1909] [Q&A]
## Qestion description
I find a question but I am not sure whether it is a bug or not. But I really want to know why.
here is the code.

```
import shapely
if __name__ == "__main__":
    a: shapely.Polygon = shapely.box(0,0,7,7)
    b: shapely.Polygon = shapely.box(1,1,10,10)
    c: shapely.Polygon = shapely.box(6,6,8,8)

    aa = shapely.intersection(a, shapely.MultiPolygon([b,c]))
```

Finally, I get the ERROR:

`
 shapely.errors.GEOSException TopologyException Input geom 1 is invalid: Nested shells at 8 6
`

I guess the reason is that **b** contains **c**, because the problem disappear when I change the code to below.

```
import shapely
if __name__ == "__main__":
    a: shapely.Polygon = shapely.box(0,0,7,7)
    b: shapely.Polygon = shapely.box(1,1,10,10)
    c: shapely.Polygon = shapely.box(6,6,8,8)

    aa = shapely.intersection(a,  shapely.MultiPolygon([shapely.unary_union([b,c])]))
```

I am curious about the reason and waiting for an good answer

## Operating system

Window 10 21H2

## Shapely version and provenance

2.0.1 installed from PyPI using pip

**Comment by theroggy** [comment]
The differenct Polygons that form a multipolygon should not intersect according to the OGC simple features specification, which is used by most GIS libraries.

Reference: the properties of a MultiPolygon are explained on page 31 of the following document (check under "Downloads"): [
Simple Feature Access – Part 1: Common Architecture](https://www.ogc.org/standard/sfa/)
